pkg/registry/common/dnsresolve/common.go

```diff
@@ -37,34 +38,48 @@ type Resolver interface {
 }
 
 func resolveDomain(ctx context.Context, service, domain string, r Resolver) (*url.URL, error) {
-	serviceDomain := fmt.Sprintf("%v.%v", service, domain)
-	_, records, err := r.LookupSRV(ctx, service, "tcp", serviceDomain)
+	var resolved string
+	if isResolved(domain) {
+		resolved = domain
+	} else {
+		serviceDomain := fmt.Sprintf("%v.%v", service, domain)
+		_, records, err := r.LookupSRV(ctx, service, "tcp", serviceDomain)
+		if err != nil {
+			return nil, err
+		}
+		if len(records) == 0 {
+			return nil, errors.New("resolver.LookupSERV return empty result")
+		}
 
-	if err != nil {
-		return nil, err
-	}
+		ips, err := r.LookupIPAddr(ctx, serviceDomain)
+		if err != nil {
+			return nil, err
+		}
+		if len(ips) == 0 {
+			return nil, errors.New("resolver.LookupIPAddr return empty result")
+		}
 
-	if len(records) == 0 {
-		return nil, errors.New("resolver.LookupSERV return empty result")
+		resolved = fmt.Sprintf("tcp://%v:%v", ips[0].IP, records[0].Port)
 	}
 
-	ips, err := r.LookupIPAddr(ctx, serviceDomain)
-
+	u, err := url.Parse(resolved)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(ips) == 0 {
-		return nil, errors.New("resolver.LookupIPAddr return empty result")
-	}
+	return u, nil
+}
 
-	u, err := url.Parse(fmt.Sprintf("tcp://%v:%v", ips[0].IP, records[0].Port))
+var resolvedPattern = regexp.MustCompile("tcp://(?P<addr>[0-9.:]+):(?P<port>[0-9]+)?")
```

Comment:
Can we just try to parse it as a URL?

Reply:
Nope, any domain is most possible a valid URL. We need a domain in `tcp://ip:port` format.

Reply:
Lets then check that part in URL after the scheme is a valid IP and PORT.

You could split Path into two strings entry to parse parts with default parsers for IP and int